src/map.js

```diff
@@ -66,99 +75,148 @@ export default class GpxMap {
+    clearScroll () {
+        this.scrolled = false;
+        this.map.addEventListener("movestart", this.markScrolled);
     }
 
     switchTheme(themeName) {
-        if (this.mapTiles)
-            this.mapTiles.removeFrom(this.map)
+        if (this.mapTiles) this.mapTiles.removeFrom(this.map);
 
-        this.mapTiles = leaflet.tileLayer.provider(themeName)
-        this.mapTiles.addTo(this.map, {detectRetina: true})
+        this.mapTiles = leaflet.tileLayer.provider(themeName);
+        this.mapTiles.addTo(this.map, {detectRetina: true});
     }
 
     updateOptions(opts) {
         if (opts.theme !== this.options.theme) {
-            this.switchTheme(opts.theme)
+            this.switchTheme(opts.theme);
+        }
+        
+        if (opts.lineOptions.existing) {
+            this.tracks.forEach(t => {
+                t.setStyle({
+                    color: opts.lineOptions.color,
+                    weight: opts.lineOptions.weight,
+                    opacity: opts.lineOptions.opacity,
+                });
+
+                t.redraw();
+            });
         }
 
-        this.tracks.forEach(t => {
-            t.setStyle({
-                color: opts.lineOptions.color,
-                weight: opts.lineOptions.weight,
-                opacity: opts.lineOptions.opacity
-            })
-
-            t.redraw()
-        })
-
-        this.options = opts
+        this.options = opts;
     }
 
     // Try to pull geo location from browser and center the map
     requestBrowserLocation() {
-        navigator.geolocation.getCurrentPosition((pos) => {
-            this.map.panTo([pos.coords.latitude, pos.coords.longitude])
-        })
+        navigator.geolocation.getCurrentPosition(pos => {
+            if (!this.scrolled && this.tracks.length == 0) {
+                this.map.panTo([pos.coords.latitude, pos.coords.longitude], {
+                    noMoveStart: true,
+                });
+                // noMoveStart doesn't seem to have an effect, see Leaflet
+                // issue: https://github.com/Leaflet/Leaflet/issues/5396
+                this.clearScroll();
+            }
+        });
     }
 
     addTrack(track) {
-        let line = leaflet.polyline(track, this.options.lineOptions)
-        line.addTo(this.map)
+        this.viewAll.enable();
+        let lineOptions = Object.assign({}, this.options.lineOptions);
+        if (lineOptions.detectColors) {
+            if (/-(Hike|Walk)\.gpx/.test(track.filename))
+                lineOptions.color = "#ffc0cb";
+            else if (/-Run\.gpx/.test(track.filename))
+                lineOptions.color = "#ff0000";
+            else if (/-Ride\.gpx/.test(track.filename))
+                lineOptions.color = "#00ffff";
+        }
+        let line = leaflet.polyline(track.points, lineOptions);
+        line.addTo(this.map);
+
+        this.tracks.push(line);
 
-        this.tracks.push(line)
+        if (!this.scrolled) this.centre();
+    }
+
+    centre() {
```

Comment:
nit: could you switch this to the American English spelling: `center`?

Reply:
fde29e987d3ef0f7fcbe7c20e63e186d65441bf7